src/internal/utils/auth.ts

```diff
@@ -0,0 +1,73 @@
+import {InvalidArgumentError} from '../../errors/errors';
+import jwtDecode from 'jwt-decode';
+import {isBase64} from './validators';
+import {decodeFromBase64} from './string';
+
+export interface LegacyClaims {
+  /**
+   * control plane endpoint
+   */
+  cp: string;
+  /**
+   * cache endpoint
+   */
+  c: string;
+}
+
+interface V1ApiTokenClaims {
+  sub: string;
+  ver: number;
+  exp: number;
+}
+
+export interface V1ApiTokenClaimsWithEndpoint extends V1ApiTokenClaims {
+  endpoint: string;
+  authToken: string;
+}
+
+export interface Base64DecodedV1Token {
+  api_key: string;
+  endpoint: string;
+}
+
+function decodeAuthTokenClaims<T>(apiToken: string): T {
+  return jwtDecode<T>(apiToken);
+}
+
+interface TokenAndEndpoints {
+  controlEndpoint: string;
+  cacheEndpoint: string;
+  authToken: string;
+}
+
+/**
+ * @param {string=} token
+ * @returns TokenAndEndpoints
+ */
+export const decodeAuthToken = (token?: string): TokenAndEndpoints => {
+  if (!token) {
+    throw new InvalidArgumentError('malformed auth token');
+  }
+
+  try {
+    if (isBase64(token)) {
```

Comment:
I wonder if a comment here about why we're checking for base64 encoding would be useful?

Reply:
added one in https://github.com/momentohq/client-sdk-nodejs/pull/336/commits/eeebeddaade055970291a058ee31981e65436625